src/main/java/org/gridsuite/actions/server/ContingencyListService.java

```diff
@@ -132,12 +132,47 @@ public Optional<PersistentContingencyList> getFormContingencyList(UUID id) {
         return self.doGetFormContingencyListWithPreFetchedCountries(id).map(ContingencyListService::fromFormContingencyListEntity);
     }
 
-    public Optional<PersistentContingencyList> getIdBasedContingencyList(UUID id) {
+    public Optional<PersistentContingencyList> getIdBasedContingencyList(UUID id, Network network) {
         Objects.requireNonNull(id);
-        return idBasedContingencyListRepository.findById(id).map(ContingencyListService::fromIdBasedContingencyListEntity);
+        return idBasedContingencyListRepository.findById(id).map(idBasedContingencyListEntity -> fromIdBasedContingencyListEntity(idBasedContingencyListEntity, network));
     }
 
-    private List<Contingency> getPowsyblContingencies(PersistentContingencyList contingencyList, UUID networkUuid, String variantId) {
+    private List<Contingency> getPowsyblContingencies(PersistentContingencyList contingencyList, Network network) {
+        ContingencyList powsyblContingencyList = contingencyList.toPowsyblContingencyList(network);
+        return powsyblContingencyList == null ? Collections.emptyList() : powsyblContingencyList.getContingencies(network);
+    }
+
+    Optional<List<ContingencyInfos>> exportContingencyList(UUID id, UUID networkUuid, String variantId) {
+        Objects.requireNonNull(id);
+        Network network = getNetworkFromUuid(networkUuid, variantId);
+        Optional<PersistentContingencyList> persistentContingencyList = getScriptContingencyList(id)
```

Comment:
You could get rid of the Optional here so you don't need to deal with it in the whole method